WalletWasabi/Helpers/PowerSaving/MacOsInhibitorTask.cs

```diff
@@ -0,0 +1,41 @@
+using System.Diagnostics;
+using WalletWasabi.Logging;
+using WalletWasabi.Microservices;
+
+namespace WalletWasabi.Helpers.PowerSaving;
+
+/// <summary>
+/// Inhibitor based on <c>caffeinate</c> command.
+/// </summary>
+public class MacOsInhibitorTask : BaseInhibitorTask
+{
+	private MacOsInhibitorTask(TimeSpan period, string reason, ProcessAsync process)
+		: base(period, reason, process)
+	{
+	}
+
+	public static MacOsInhibitorTask Create(TimeSpan basePeriod, string reason)
+	{
+		string innerCommand = $$"""
+			caffeinate -i &
+			caffeinatePid=\$!;
+			trap \"kill -9 \$caffeinatePid\" 0 SIGINT SIGTERM;
+			while [ -n "$(grep {{Environment.ProcessId}})" ];
+			do
+			    sleep 1;
+			done;
+			""".ReplaceLineEndings(replacementText: " ");
+
```

Comment:
```suggestion
		string innerCommand = $$"""
			caffeinate -i &
			caffeinatePid=$!;
			trap \"kill -9 \$caffeinatePid\" 0 SIGINT SIGTERM
			while [ -n \"$(grep {{ Environment.ProcessId}}     )\" ]
			do
				sleep 1;
			done;
			""" .ReplaceLineEndings(replacementText: " ");

```

All other variants from this had issues, this suggestions work iin all cases (standard stop and SIGKILL wasabi)

Reply:
> no semicolon after `trap \"kill -9 \$caffeinatePid\" 0 SIGINT SIGTERM`?

I temporarily failed but quickly edited, you caught it fast 😆 
Anyways you pushed the good version.